src/lematerial_forgebench/metrics/base.py

```diff
@@ -432,7 +451,10 @@ def __call__(
             Object containing the metric value and computation metadata.
         """
         structures_list = format_structures(structures)
-        return self.compute(structures_list)
+        if reference_df is None:
```

Comment:
Isn't this redundant with the subsequent checks? `self.compute` can handle a `None` `reference_df`